src/chatgpt.py

```diff
@@ -302,10 +365,22 @@ def start_prompt(
         raise KeyboardInterrupt
 
     messages.append({"role": "user", "content": message})
+    if config["supplier"] == "azure":
+        api_key = config["AZURE_OPENAI_API_KEY"]
+        # Base body parameters
+        model = config["AZURE_OPENAI_DEPLOYMENT_NAME"]
+        api_version = config['AZURE_OPENAI_API_VERSION']
+        base_endpoint = config["AZURE_OPENAI_ENDPOINT"]
+    elif config["supplier"] == "openai":
+        api_key = config["OPENAI_API_KEY"]
+        model = config["OPENAI_MODEL"]
+        base_endpoint = config["OPENAI_ENDPOINT"]
+    else:
+        raise ValueError("Supplier must be either 'azure' or 'openai'")
```

Comment:
I wonder if it makes sense at this point to create a `Supplier` class and then have `OpenAISupplier` and `AzureOpenAISupplier` subclasses? Otherwise this could get long and hard to maintain over time.